Networking/NetworkingTests/Remote/ProductsRemoteTests.swift

```diff
@@ -583,6 +583,34 @@ final class ProductsRemoteTests: XCTestCase {
         XCTAssertTrue(result.isFailure)
     }
 
+    // MARK: - Products batch update
+
+    /// Verifies that updateProducts properly parses the `products-batch-update` sample response.
+    ///
+    func test_bulk_update_products_properly_returns_parsed_products() {
+        // Given
+        let remote = ProductsRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "products/batch", filename: "products-batch-update")
+
+        // When
+        let sampleProducts = [sampleProduct()]
+        var updatedProducts: [Product]?
+        waitForExpectation { expectation in
+            remote.updateProducts(siteID: self.sampleSiteID, products: sampleProducts) { result in
+                // Then
+                guard case let .success(products) = result else {
+                    XCTFail("Unexpected result: \(result)")
+                    return
+                }
+                updatedProducts = products
+                expectation.fulfill()
+            }
+        }
```

Comment:
nit: Rather than using an expectation here, we could assign `updatedProducts` directly with `waitFor` and a promise. For example:

```suggestion
        var updatedProducts: [Product] = waitFor { promise in
            remote.updateProducts(siteID: self.sampleSiteID, products: sampleProducts) { result in
                // Then
                guard case let .success(products) = result else {
                    XCTFail("Unexpected result: \(result)")
                    return
                }
                promise(products)
            }
        }
```

Reply:
Good point! Updated in 18cad51.